### ISSUE TITLE
Function argument prop type mismatch is reported as another, enum prop type mismatch

### ISSUE THREAD

**Issue by growtofill** [microsoft/TypeScript#35666]
**TypeScript Version:**  3.8.0-dev.20191212

**Search Terms:**
type enum is not assignable to type enum member

**Code**

```ts
enum State {
    SOME,
    NONE,
}

interface StateValue {
    state: State.SOME
    value: boolean
}

const stateValue = {
    state: State.SOME,
    value: 0
}

function f(_: StateValue) {}

f(stateValue) // error

```

**Expected behavior:**
```
Type of property 'value' is incompatible.
```

**Actual behavior:**
```
Argument of type '{ state: State; value: number; }' is not assignable to parameter of type 'StateValue'.
  Types of property 'state' are incompatible.
    Type 'State' is not assignable to type 'State.SOME'.(2345)
```

**Playground Link:** 
http://www.typescriptlang.org/play/?ts=3.8.0-dev.20191212#code/KYOwrgtgBAygLgQzsKBvAUFLsDyBZAUQBpNsA5HM49AX3XQEsRkAnAMwQGMV4lgA1BABswKDNigBnRMgBcsGcAB0MfAVJYAbsNHyARgHsDQ4AhC16nAyGlTFgkSgC8aDXb7zeyFWpITtjvIADBbobGAgnHAM1lBsABQA+p72OsAAlGh0YfHSfA6i6ehAA

**Related Issues:** 
Nothing I was able to find

**Comment by ilogico** [comment]
That's normal, like in `const x = { v: 'something' };` where `x` has type `{ v: string }` and not `{ v: 'something' }`.
You can fix it by using type annotations: `const stateValue: StateValue = ...`
By using casts: `const stateValue = { state: State.SOME as State.SOME, ... }`
Or by using `as const` (not sure if this is also considered a cast): `const stateValue = { state: State.SOME as const, ... }`

**Comment by growtofill** [comment]
@ilogico Makes sense, although I can't get used to `State.SOME as State.SOME`.

Didn't know about `as const`, thank you, Gonçalo!
